Makefile

```diff
@@ -112,6 +130,20 @@ cluster-sync-handler: cluster-sync-resources $(local_handler_manifest)
 
 cluster-sync: cluster-sync-handler
 
+$(description): version/description
+	mkdir -p $(dir $@)
+	sed "s#HANDLER_IMAGE#$(HANDLER_IMAGE)#" \
+		version/description > $@
+
+release: HANDLER_IMAGE_SUFFIX = :$(shell hack/version.sh)
```

Comment:
Please explain how this whole mechanism works. Having all the logic in Makefile without any comments is unmaintaintable and unreviewable.

Reply:
Will do a bigger PR with documentation on it.